dll/win32/shell32/dialogs/dialogs.cpp

```diff
@@ -324,6 +324,12 @@ INT_PTR CALLBACK PickIconProc(
                 case ODA_SELECT:
                 case ODA_DRAWENTIRE:
                 {
+                    // Do not draw hidden icons
+                    UINT Visible = SendMessageW(pIconContext->hDlgCtrl, LB_GETLISTBOXINFO, 0, 0);
```

Comment:
Does this really optimize anything? Looks to me like additional processing for the most common case to "optimize" a corner case.

Reply:
If we load the shell32.dll file, which contains 232 icons, we save (Item_Count - Top_Item - Visible_Items) calls to the DrawIconEx (...) function.
If the first item in the list is index 50, we eliminate 232-50-28 = 154 superfull calls to DrawIconEX (...).
There will certainly be more effective forms of optimization, but this is already noticeable in every refresh of the listbox.